tomopy/misc/corr.py

```diff
@@ -375,21 +377,74 @@ def remove_outlier(arr, dif, size=3, axis=0, ncore=None, out=None):
 
     tmp = np.empty_like(arr)
 
-    if ncore is None:
-        ncore = mproc.mp.cpu_count()
+    ncore, chnk_slices = mproc.get_ncore_slices(arr.shape[axis], ncore=ncore)
+    
+    filt_size = [size]*arr.ndim
+    filt_size[axis] = 1
 
     with cf.ThreadPoolExecutor(ncore) as e:
         slc = [slice(None)]*arr.ndim
-        for i in range(arr.shape[axis]):
-            slc[axis] = i
-            e.submit(filters.median_filter, arr[slc], size=(size, size),
+        for i in range(ncore):
+            slc[axis] = chnk_slices[i]
+            e.submit(filters.median_filter, arr[slc], size=filt_size,
                      output=tmp[slc])
 
     with mproc.set_numexpr_threads(ncore):
         out = ne.evaluate('where(arr-tmp>=dif,tmp,arr)', out=out)
 
     return out
 
+def remove_outlier1d(arr, dif, size=3, axis=0, ncore=None, out=None):
+    """
+    Remove high intensity bright spots from an array, using a one-dimensional
```

Comment:
I think the method works differently if I remember correctly: it computes the 1-D median filter along the axis defined by the axis argument, and slices along a different axis (the largest of the remaining axes). The other outlier methods indeed do the slicing in the direction of the axis argument, and perform the filtering along the other axes - this was confusing to me in the old method documentation (I only understood it by reading the code), so the PR also includes a clarification of this in the documentation.